### DISCUSSION TITLE
Staging environment metrics

### DISCUSSION THREAD

**Discussion by RIshabh9727** [mulesoft-catalyst/metrics-toolkit#154] [Q&A] [answered]
Hi team,

I have **three environments** in my mulesoft anypoint platform-- staging & sandbox of type non-prod & production of type prod. But in the output of platform metrics, I can see **metrics** only for **sandbox & production**. Can someone please help me with how to get stats for the **staging environment**?

Regards,
Rishabh

**Comment by richardmckinley** [comment]
Hi @RIshabh9727, 

It depends on what loader you are using to extract metrics. For example, if you use the JSON loader, with raw=false you will get nicely formatted environment level details. If you set raw=true you will get less formatted but more detailed information at the environment level. 

If you are using the CSV output, please note that metrics are provided at the environment type level i.e. Sandbox or Production, not the environment level. We are currently working on how best to provide environment level metrics in the CSV output - see #149 

I hope that helps clarify. If you have more questions please let me know. 

Richard

**Reply by RIshabh9727** [reply]
Hy @richardmckinley 

Thanks for responding. I got a little bit confused, I am using the loader strategy as **splunk**. So in Splunk is it possible to fetch metrics for individual environments. Like in my case I want it for all three env. -- production, staging & sandbox. 

Many thanks
Rishabh

**Comment by RIshabh9727** [comment]
Hy @richardmckinley 

I need one more help, as I can see we have default dashboards for sandbox & production only, Can you suggest to me what changes I need to make in the existing dashboard to make it suitable for the staging environment.

**Comment by richardmckinley** [comment, marked as answer]
Hi @RIshabh9727 
I'm afraid I'm not a Splunk expert so cannot easily provide a query filtering metrics by environment. However, as the information is present in the events it should be possible with some knowledge of Splunk queries. The Splunk (and other) dashboards are provided as an initial view of what metrics can be displayed, rather than a comprehensive set of views. 

If you figure out a query to filter by environment it would be great if you could share so others would also be able to leverage it. 

Additionally, if you think this is something we should invest time in adding to the dashboards, please fee free to raise this via the issues feature. 

Thanks and regards, 
Richard
